mlflow/pyfunc/utils/input_converter.py

```diff
@@ -1,15 +1,24 @@
 from dataclasses import fields, is_dataclass
-from typing import get_args, get_origin
+from typing import Union, get_args, get_origin
 
 from mlflow.utils.annotations import experimental
 
 
+def _is_optional_dataclass(field_type) -> bool:
+    """Check if the field type is an Optional containing a dataclass."""
+    if get_origin(field_type) is Union:
+        inner_types = get_args(field_type)
+        # Check if it's a Union[Dataclass, NoneType] (i.e., Optional[Dataclass])
+        if len(inner_types) == 2 and inner_types[1] == type(None):
+            effective_type = next(t for t in get_args(field_type) if t is not type(None))
+            return is_dataclass(effective_type)
+    return False
```

Comment:
Does this work if the type hint looks like: `a: int | None`. This is more common in Python >= 3.10.

Reply:
hm this does not work because the `| operator` turns it into a

`types.UnionType` instead of a `typing.Union` type

do you have any advice here on making this code work for both cases?

Reply:
is there a way to get the original type hint as string?

Reply:
discussed offline: will not support the new 3.10 notation for now, left a comment describing as much